hpx/runtime/applier/apply_helper.hpp

```diff
@@ -116,39 +115,78 @@ namespace hpx { namespace applier { namespace detail
         }
     };
 
+    ///////////////////////////////////////////////////////////////////////////
+    HPX_FORCEINLINE bool needs_to_execute_asynchronously()
```

Comment:
Yah, it's probably mainly the amount of stack which has to be available for going ahead synchronously. I'll factor out the function.